example/build.sbt

```diff
@@ -40,6 +38,4 @@ scalacOptions ++= Seq(
   lintFlags.value
 )
 
-addCompilerPlugin("io.tryp" % "splain" % "0.5.8" cross CrossVersion.patch)
```

Comment:
@jcazevedo I'm removing splain here because it's obsolete - it was integrated in Scala 2.13.6 compiler behind a compiler flag (https://github.com/tek/splain/issues/53). We probably should update the "implicit not found" docs to reference these new flags.

Reply:
We might want to keep it in the example project for Scala 2.12, since it seems like it's being backported there. We can keep it out while there's no version available for Scala 2.12.14.

Reply:
I'd actually prefer to keep it out permanently, even if a backport was released. My main concern is how these plugin dependencies affect our ability to adopt new Scala versions: between scoverage, scalafix, scalafmt and now splain, it's becoming increasingly difficult to move fast when a new version is released.

Another argument against it is that I'd like to keep our example project simple, including the build - and our build.sbt there is already a bit more complex than I'd like, with us choosing the PureConfig version to use dynamically and keeping per-version lint flags to mimic the main project.